### PR TITLE
Update Service Account Queries/Mutations

### PR DESCRIPTION
**Pull request by andriisoldatenko**
related PR https://github.com/astronomer/houston-api/pull/137/files
issue https://github.com/astronomer/issues/issues/349

TODO:
- fixed all go fmt issues.
- added latest release to README.md

### CODE REVIEW

File: cmd/deployment.go [hunk 1]
```diff
@@ -155,3 +192,17 @@ func deploymentUpdate(cmd *cobra.Command, args []string) error {
...
+	}
+
+	if err := validateRole(role); err != nil {
+		return errors.Wrap(err, "failed to find a valid role")
+	}
+	fullRole := strings.Join([]string{"DEPLOYMENT", strings.ToUpper(role)}, "_")
+	// Silence Usage as we have now validated command input
+	cmd.SilenceUsage = true
+	return sa.CreateUsingDeploymentUUID(deploymentUuid, label, category, fullRole)
+}
```
Comment (schnie) [inline cmd/deployment.go]: @andriisoldatenko newline here

Comment (andriisoldatenko) [reply cmd/deployment.go]: fixed.

**Comment by andriisoldatenko** [comment]
@schnie pls review
```
astro sa
Service-accounts represent a revokable token with access to the Astronomer platform

Usage:
  astro service-account [command]

Aliases:
  service-account, sa

Available Commands:
  delete      Delete a service-account in the astronomer platform
  get         Get a service-account by entity type and entity id

Flags:
  -h, --help   help for service-account

Use "astro service-account [command] --help" for more information about a command.
```

```
astro deployment service-account create --deployment-uuid=ck1qg6zv3002f0869ypx5a4ds --label=my_label --role=viewer
 NAME         CATEGORY     ID                            APIKEY
 my_label     default      ck1qk2kfn008u0869boqqi6cd     00000000000000111111111

 Service account successfully created.
```

```
astro workspace service-account create --workspace-uuid=ck1qg6whg001r08691y117hub --label=my_label --role=viewer
 NAME         CATEGORY     ID                            APIKEY
 my_label     default      ck1qiir3b006t0869gk58lnl7     111111122222222

 Service account successfully created.
```

**Comment by schnie** [comment]
@andriisoldatenko should we remove the old `astro service-account` command?

**Comment by andriisoldatenko** [comment]
@schnie i did in 309b609a7a24ec3b9321b19218c825c9da44bb90

**Comment by andriisoldatenko** [comment]
or do you mean to move `astro sa get/delete` to `astro workspace sa get/delte`?

**Comment by schnie** [comment]
Yea, to line up with our mutation/query schema in Houston, I feel like everything should be nested under `astro workspace sa` or `astro deployment sa`

**Comment by andriisoldatenko** [comment]
@schnie 
>Yea, to line up with our mutation/query schema in Houston, I feel like everything should be nested under astro workspace sa or astro deployment sa

do we want to show depreciation warning when user run:
`astro service-account` ? or just show:
```
Error: unknown command "service-account" for "astro"
Run 'astro --help' for usage.
```

**Comment by schnie** [comment]
@andriisoldatenko a deprecation notice would be nice. Can we remove the `astro service-account` command from the help text, but show the notice if they do run the old command?

Also looks like we got some conflicts

**Comment by andriisoldatenko** [comment]
@schnie @ashb PTAL

**Comment by andriisoldatenko** [comment]
New astro root output (wo service-account):
```
astro is a command line interface for working with the Astronomer Platform.

Usage:
  astro [command]

Available Commands:
  auth            Manage astronomer identity
  cluster         Manage Astronomer EE clusters
  completion      Generate autocompletions script for the specified shell (bash or zsh)
  config          Manage astro project configurations
  deploy          Deploy an airflow project
  deployment      Manage airflow deployments
  dev             Manage airflow projects
  help            Help about any command
  upgrade         Check for newer version of Astronomer CLI
  user            Manage astronomer user
  version         Astronomer CLI version
  workspace       Manage Astronomer workspaces

Flags:
  -h, --help   help for astro

Use "astro [command] --help" for more information about a command.
```

**Comment by schnie** [comment]
@andriisoldatenko is this good to go?

**Comment by andriisoldatenko** [comment]
@schnie code is ready, 
i've tested `astro workspace service-account create/get/delete` it works! 
But it would be great to test `astro deployment service-account create/get/delete`.

**Comment by andriisoldatenko** [comment]
added few more tests, ready to merge.

**Pull request merged**